store/query/auth.js

```diff
@@ -1,11 +1,23 @@
 const passwordHash = require('../util/passwordHash');
+const moment = require('moment');
 
 module.exports = async function(email, password) {
 	const hash = passwordHash(password);
+	const user = await this.findOne({ email });// the user has to be checked for locked whether the password was correct or not.
 
-	const users = await this.find({ email, password: hash });
-	if (users.length === 0) throw new Error('Wrong e-mail or password');
+	if (user.locked && moment().isBefore(user.locked)) {
+		throw { Error: 'LockError', retry: user.locked };
```

Comment:
Change the key to `name` instead of `Error`. A true error object has `{ name: 'ErrorType', message: 'details' }`. Sticking to this convention will allow us to replace this object literal with an error instance.